pkg/typesystem/typesystem.go

```diff
@@ -850,6 +839,7 @@ func NewAndValidate(ctx context.Context, model *openfgav1.AuthorizationModel) (*
 			if err != nil {
 				return nil, err
 			}
+			t.AssignTerminalTypes(typeName, relationName)
```

Comment:
Easy to miss in this PR but the `NewAndValidate` function is where the connected types mapping occurs. The `NewAndValidate` is memoized and is only called once per store operation per node lifetime.